### PR TITLE
Add a ctest using letkfoi to assimilate GTS snow depth.

### PR DESCRIPTION
**Pull request by jiaruidong2017**
Add a ctest by using JEDI letkfoi to assimilate the GTS snow depth (Issue #241). 
To avoid staging too much data (two ensemble members), we use the previous ctest (`test_gdasapp_land_create_ens`) to generate two required ensemble members for use in this ctest. 

One obs data is required for this test as: 

COPY the data: 
`/scratch2/NCEPDEV/land/data/DA/snow_depth/GTS/data_proc/202103/adpsfc_snow_2021032318.nc4`
TO the directory at:
`/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/data/land/snow_depth/GTS/202103/`

### CODE REVIEW

File: test/land/letkfoi_snowda.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+mkdir -p Data diags
+ln -s ${project_binary_dir}/fv3-jedi/test/Data/fieldmetadata Data/fieldmetadata
+ln -s ${project_binary_dir}/fv3-jedi/test/Data/fv3files Data/fv3files
+ln -s ${project_source_dir}/ush/land/genYAML_output_letkfoi.yaml letkf_land.yaml
+ln -s /scratch2/NCEPDEV/land/data/DA/snow_depth/GTS/data_proc/202103/adpsfc_snow_2021032318.nc4 adpsfc_snow.nc4
+#ln -s ${OBSDIR}/snow_depth/GTS/202103/adpsfc_snow_2021032318.nc4 adpsfc_snow.nc4
+ln -s ${OBSDIR} Data/land
+echo 'do_landDA: calling fv3-jedi'
+
+srun '--export=ALL' -n 6 ${EXECDIR}/${JEDI_EXEC} letkf_land.yaml ${WORKDIR}/letkfoi_snowda.log
```
Comment (CoryMartin-NOAA) [inline test/land/letkfoi_snowda.sh]: Does this log file get populated correctly? Wouldn't you need to pipe stdout to it? Or can OOPS just take a logfile as an argument?

File: test/land/letkfoi_snowda.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+fi
+############################################
+# Prepare and Run JEDI
+############################################
+mkdir -p Data diags
+ln -s ${project_binary_dir}/fv3-jedi/test/Data/fieldmetadata Data/fieldmetadata
+ln -s ${project_binary_dir}/fv3-jedi/test/Data/fv3files Data/fv3files
+ln -s ${project_source_dir}/ush/land/genYAML_output_letkfoi.yaml letkf_land.yaml
+ln -s /scratch2/NCEPDEV/land/data/DA/snow_depth/GTS/data_proc/202103/adpsfc_snow_2021032318.nc4 adpsfc_snow.nc4
+#ln -s ${OBSDIR}/snow_depth/GTS/202103/adpsfc_snow_2021032318.nc4 adpsfc_snow.nc4
```
Comment (CoryMartin-NOAA) [inline test/land/letkfoi_snowda.sh]: Remove if unneeded

File: test/land/letkfoi_snowda.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+fi
+############################################
+# Prepare and Run JEDI
+############################################
+mkdir -p Data diags
+ln -s ${project_binary_dir}/fv3-jedi/test/Data/fieldmetadata Data/fieldmetadata
+ln -s ${project_binary_dir}/fv3-jedi/test/Data/fv3files Data/fv3files
+ln -s ${project_source_dir}/ush/land/genYAML_output_letkfoi.yaml letkf_land.yaml
+ln -s /scratch2/NCEPDEV/land/data/DA/snow_depth/GTS/data_proc/202103/adpsfc_snow_2021032318.nc4 adpsfc_snow.nc4
```
Comment (CoryMartin-NOAA) [inline test/land/letkfoi_snowda.sh]: No hard coded paths, do I need to copy this to the staged test data area on Hera/Orion? Or is it already there? Or is it generated by one of the earlier tests?

**Review by CoryMartin-NOAA — COMMENTED** [review]
3 minor comments

File: test/land/letkfoi_snowda.sh (hunk 2 above)
Comment (jiaruidong2017) [reply test/land/letkfoi_snowda.sh]: When you copy this GTS obs `adpsfc_snow_2021032318.nc4` to your directory at `/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/data/land/snow_depth/GTS/202103/`, we will use this commented line and remove the line above.

Comment (CoryMartin-NOAA) [reply test/land/letkfoi_snowda.sh]: Got it, I missed the description in the PR, I see it now. Copied on Hera, will rsync to Orion right now.

File: test/land/letkfoi_snowda.sh (hunk 3 above)
Comment (jiaruidong2017) [reply test/land/letkfoi_snowda.sh]: As mentioned above, we will remove this line and replace this line with the commented line. This data is not generated by the earlier tests, and the data need to be copied to your directory at `/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/data/land/snow_depth/GTS/202103/`.

File: test/land/letkfoi_snowda.sh (hunk 1 above)
Comment (jiaruidong2017) [reply test/land/letkfoi_snowda.sh]: I checked my working directory and found these logfiles were populated correctly. I believe to keep these logfiles just for the debugging purpose. I will remove this log file from the command lines.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**